nextcord/application_command.py

```diff
@@ -0,0 +1,1266 @@
+    def parse_discord_response(self, state: ConnectionState, data: dict) -> None:
+        self.set_state(state)
+        # if guild_id:
+        command_id = int(data["id"])
+        if guild_id := data.get("guild_id", None):
+            # self._guild_command_ids[guild_id] = command_id
+            self._command_ids[guild_id] = command_id
+            self._guild_ids.add(guild_id)
+            self.add_guild_rollout(guild_id)
+        else:
+            # self._global_command_id = command_id
+            self._command_ids[None] = command_id
+
+    def _get_basic_application_payload(self) -> dict:
+        """Bare minimum payload that both Global and Guild commands can use."""
+        payload = super().payload
+        if self.type is not ApplicationCommandType.chat_input and "options" in payload:
+            payload.pop("options")
+        if self.default_permission is not None:
+            payload["default_permission"] = self.default_permission
+        return payload
+
+    def _handle_resolved_message(self, message_data: dict) -> Message:
+        # TODO: This is garbage, find a better way to add a Message to the cache.
+        #  It's not that I'm unhappy adding things to the cache, it's having to manually do it like this.
+        # The interaction gives us message data, might as well use it and add it to the cache.
+        channel, guild = self._state._get_guild_channel(message_data)
+        message = Message(channel=channel, data=message_data, state=self._state)
+        if not self._state._get_message(message.id) and self._state._messages is not None:
+            self._state._messages.append(message)
+        return message
+
+    def _handle_resolved_user(self, user_data: dict) -> User:
+        """Takes the raw user data payload from Discord and adds it to the state cache."""
+        # This is what I prefer _handle_resolved_message to look like.
+        return self._state.store_user(user_data)
+
+    @property
+    def payload(self) -> List[dict]:
+        """Returns a list of Discord "Application Command Structure" payloads.
+
+        Returns
+        -------
+        payloads: List[:class:`dict`]
+            Returns a list containing the global command payload, if enabled, and payloads corresponding to every guild
+            ID specified.
+        """
+        # TODO: This always returns a list, should it be "payloads"? Won't override subcommand payload though.
+        partial_payload = self._get_basic_application_payload()
+
+        ret = []
+        if self.is_guild:
+            for guild_id in self.guild_ids:
+                temp = (partial_payload.copy())  # We don't need to make a deep copy as guild_id is on the top layer.
+                temp["guild_id"] = guild_id
+                ret.append(temp)
+        if self.is_global:
+            ret.append(partial_payload)
+        return ret
+
+    def check_against_raw_payload(self, raw_payload: dict, guild_id: Optional[int]) -> bool:
```

Comment:
``guild_id`` is marked optional, but a guild id is required here? Default to ``None``